pkg/pgmodel/pgx.go

```diff
@@ -557,25 +591,38 @@ func (h *insertHandler) flush() {
 	if !h.hasPendingReqs() {
 		return
 	}
-	h.flushPending(h.pending)
+	h.flushPending()
 }
 
-func (h *insertHandler) flushPending(pending *pendingBuffer) {
-	err := func() error {
-		_, err := h.setSeriesIds(pending.batch.sampleInfos)
-		if err != nil {
-			return err
-		}
+func (h *insertHandler) flushPending() {
+	_, err := h.setSeriesIds(h.pending.batch.sampleInfos)
```

Comment:
hmm, what's the reason this is done here and not by the copiers?

Reply:
@cevian the series cache needs to be per-metric-inserter, while the copiers are multiplexed across all inserters.